### PR TITLE
add loongarch64 support for open-build-service

### PR DESCRIPTION
**Pull request by zhangwenlong8911**

### CODE REVIEW

**Comment by adrianschroeter** [comment]
pushed minor fixes and a data migration to add the arch.

I am also loonking into the build script, which seems to have incomplete support for loongarch64 atm. Did you build anything successfully using it already?
Do you know, if kvm already supported on loonarch64?

**Comment by zhangwenlong8911** [comment]
> pushed minor fixes and a data migration to add the arch.
> 
> I am also loonking into the build script, which seems to have incomplete support for loongarch64 atm. Did you build anything successfully using it already? Do you know, if kvm already supported on loonarch64?

I can build with a low version and use it in some projects, such as http://117.78.1.88/project/show/openEuler:22.03:LTS:LoongArch

kvm does not fully support loongarch64, but support work is in progress in an orderly manner, see the mailing list: lkml.org/lkml/2023/4/6/1613

**Review by hennevogel — APPROVED** [review]
for the frontend changes...

File: src/api/db/data/20230411174510_add_loongarch64.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddLoongarch64 < ActiveRecord::Migration[5.1]
```
Comment (eduardoj) [inline src/api/db/data/20230411174510_add_loongarch64.rb:1]:
```suggestion
class AddLoongarch64 < ActiveRecord::Migration[7.0]
```
This seems copy-pasted from other data migrations. I tested the data migration generator (`bin/rails generate data_migration ...`). It returned `ActiveRecord::Migration[7.0]`.

Is there any reason to not using the now 7.0 version default?

**Review by eduardoj — CHANGES_REQUESTED** [review]

File: src/api/db/data/20230411174510_add_loongarch64.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  def self.up
```
Comment (eduardoj) [inline src/api/db/data/20230411174510_add_loongarch64.rb:2]:
```suggestion
  def up
```

**Comment by zhangwenlong8911** [comment]
@eduardoj Does this pr still need to be revised, please help review the pr if you have time, thank you

**Comment by hennevogel** [comment]
@adrianschroeter is this good to go (after rebase)?

**Comment by zhangwenlong8911** [comment]
@eduardoj @adrianschroeter please help review the pr if you have time, thank you

**Pull request merged**